fix debug_info arg in createOperator and improve reroute_tensor

harouwu opened the pull request:
Summary:
-Fixed C2 core.CreateOperator debug info assignment
-Improving core.Net.reroute_tensor

Differential Revision: D9426659

Maratyszcza reviewed (approved):
LG. Not an expert in this part of Caffe2, though

harouwu committed:
fix debug_info arg in createOperator and improve reroute_tensor (#10736)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/10736

-Fixed C2 core.CreateOperator debug info assignment
-Improving core.Net.reroute_tensor

Differential Revision: D9426659

fbshipit-source-id: a299427acb96800b8c5ea5018c140634c108f2e4
